### ISSUE TITLE
Server crashes when throwing inside Cloud Code trigger

### ISSUE THREAD

**Issue by mtrezza** [parse-community/parse-server#8255]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description

Parse Server crashes when throwing inside a Cloud Code trigger. The crash is due to the removal of the `UnhandledPromiseRejectionWarning` in Node 15. Parse Server handles trowing inside a Cloud Code function differently and returns a Parse error without crashing the server.

#### Breaking Change

This change is breaking because developers may have implemented their own solution by globally catching unhandled rejections on the node process with:

```
process.on('unhandledRejection', (reason, promise) => {
  // handle error
});
```

### Steps to reproduce

This test fails:
```js
it('does not crash server when throwing in afterLogout hook', async () => {
    let triggered = false;
    const error = new Error('throw in cloud code');

    Parse.Cloud.afterLogout(() => {
      triggered = true;
      throw error;
    });

    await Parse.User.signUp('user', 'pass');
    const response = await Parse.User.logOut().catch(e => e);
    expect(triggered).toBeTrue();
    expect(response).toBeDefined();
    expect(response.code).toBe(141);
    expect(response.message).toBe(error.message);
  });
```

However, throwing in a Cloud Code function passes the test:

```js
it('does not crash server when throwing in cloud code function', async () => {
    const error = new Error('throw in cloud code');
    Parse.Cloud.define('hello', () => {
      throw error;
    });

    const response = await Parse.Cloud.run('hello').catch(e => e);
    expect(response).toBeDefined();
    expect(response.code).toBe(141);
    expect(response.message).toBe(error.message);
  });
```

### Actual Outcome

Throwing in trigger crashes server.

### Expected Outcome

Throwing in trigger returns Parse error without crashing the server.

### Environment

Server
- Parse Server version: [5.3.0-alpha.30](https://github.com/parse-community/parse-server/releases/tag/5.3.0-alpha.30)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0-alpha.5](https://github.com/parse-community/parse-server/releases/tag/6.0.0-alpha.5)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/6.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0](https://github.com/parse-community/parse-server/releases/tag/6.0.0)
